src/Token/Parser.php

```diff
@@ -90,7 +91,30 @@ private function parseHeader(string $data): array
      */
     private function parseClaims(string $data): array
     {
-        return (array) $this->decoder->jsonDecode($this->decoder->base64UrlDecode($data));
+        $claims = (array) $this->decoder->jsonDecode($this->decoder->base64UrlDecode($data));
+
+        if (isset($claims[RegisteredClaims::AUDIENCE])) {
+            $claims[RegisteredClaims::AUDIENCE] = (array) $claims[RegisteredClaims::AUDIENCE];
+        }
+
+        foreach (RegisteredClaims::DATE_CLAIMS as $claim) {
+            if (!isset($claims[$claim])) {
```

Comment:
In order to use it I would also have to do an `array_flip(RegisteredClaims::DATE_CLAIMS)` (or an `array_keys($claims)` and use `array_intersect()` instead) not sure if will help us...